src/zarr/storage/object_store.py

```diff
@@ -38,18 +38,29 @@ def __eq__(self, value: object) -> bool:
         if not isinstance(value, ObjectStore):
             return False
 
-        return self.store.__eq__(value.store)
+        return bool(self.store.__eq__(value.store))
 
     def __init__(self, store: _ObjectStore, *, read_only: bool = False) -> None:
+        if not isinstance(
+            store,
+            (
+                obs.store.AzureStore,
+                obs.store.GCSStore,
+                obs.store.HTTPStore,
+                obs.store.S3Store,
+                obs.store.LocalStore,
+                obs.store.MemoryStore,
+            ),
+        ):
```

Comment:
@kylebarron I couldn't find an abstract base class or another way to simplify this, but suggestions for better approaches would be welcome

Reply:
Hmm, perhaps we should have a base class.